pkg/archive/path_unix.go

```diff
@@ -2,8 +2,57 @@
 
 package archive
 
+import (
+	"fmt"
+	"io/fs"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
 // checkSystemDriveAndRemoveDriveLetter is the non-Windows implementation
 // of CheckSystemDriveAndRemoveDriveLetter
 func checkSystemDriveAndRemoveDriveLetter(path string) (string, error) {
 	return path, nil
 }
+
+// lgetxattr retrieves the value of the extended attribute identified by attr
+// and associated with the given path in the file system.
+// It returns a nil slice and nil error if the xattr is not set.
+func lgetxattr(path string, attr string) ([]byte, error) {
+	// Start with a 128 length byte array
+	dest := make([]byte, 128)
+	sz, err := unix.Lgetxattr(path, attr, dest)
+
+	for errors.Is(err, unix.ERANGE) {
+		// Buffer too small, use zero-sized buffer to get the actual size
+		sz, err = unix.Lgetxattr(path, attr, []byte{})
```

Comment:
More failures in the buildx repository, which also runs on OpenBSD (looks like xattrs are not supported there); 

- https://github.com/docker/buildx/pull/2868

```
#25 [linux/amd64->openbsd/amd64 buildx-build 1/1] RUN --mount=type=bind,target=.   --mount=type=cache,target=/root/.cache   --mount=type=cache,target=/go/pkg/mod   --mount=type=bind,from=buildx-version,source=/buildx-version,target=/buildx-version <<EOT (set -e...)
#25 0.070 + CGO_ENABLED=0 go build -mod vendor -trimpath -ldflags '-X github.com/docker/buildx/version.Version=53dca9a -X github.com/docker/buildx/version.Revision=53dca9a8946feca5ba7a63cb96140d94f2067656 -X github.com/docker/buildx/version.Package=github.com/docker/buildx -s -w' -o /usr/bin/docker-buildx ./cmd/buildx
#25 35.10 # github.com/docker/docker/pkg/archive
#25 35.10 vendor/github.com/docker/docker/pkg/archive/path_unix.go:25:18: undefined: unix.Lgetxattr
#25 35.10 vendor/github.com/docker/docker/pkg/archive/path_unix.go:29:18: undefined: unix.Lgetxattr
#25 35.10 vendor/github.com/docker/docker/pkg/archive/path_unix.go:34:18: undefined: unix.Lgetxattr
#25 35.10 vendor/github.com/docker/docker/pkg/archive/path_unix.go:38:26: undefined: unix.ENODATA
#25 35.10 vendor/github.com/docker/docker/pkg/archive/path_unix.go:50:53: undefined: unix.Lsetxattr
```

Reply:
Previously this code was in `_linux.go` file, so the xattrs were not used on BSD before. I think we can leave making the BSD work for a follow up.

Reply:
Updated, it should match what is defined in `golang.org/x/sys/unix` now